src/test/java/io/orkes/conductor/client/e2e/TaskQueuePermissionTests.java

```diff
@@ -0,0 +1,80 @@
+package io.orkes.conductor.client.e2e;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+import java.util.Map;
+
+import org.junit.jupiter.api.Test;
+
+import com.netflix.conductor.common.metadata.workflow.StartWorkflowRequest;
+
+import io.orkes.conductor.client.MetadataClient;
+import io.orkes.conductor.client.TaskClient;
+import io.orkes.conductor.client.WorkflowClient;
+import io.orkes.conductor.client.http.OrkesTaskClient;
+import io.orkes.conductor.client.model.UpsertGroupRequest.RolesEnum;
+import io.orkes.conductor.client.util.RegistrationUtil;
+import io.orkes.conductor.client.util.authentication.TestWithAuthentication;
+
+public class TaskQueuePermissionTests extends TestWithAuthentication {
+
+    private static final int RETRY_LIMIT = 5;
+
+    private final WorkflowClient workflowClient;
+    private final MetadataClient metadataClient;
+
+    public TaskQueuePermissionTests() {
+        workflowClient = orkesClients.getWorkflowClient();
+        metadataClient = orkesClients.getMetadataClient();
+    }
+
+    @Test
+    public void testTaskQueuePermissionForUserWithPreAuthorizedRole() {
+        var adminOrkesClients = getOrkesClientsWithRole(RolesEnum.ADMIN);
```

Comment:
We already have a way to get the admin client, https://github.com/orkes-io/orkes-conductor-client/blob/main/src/test/java/io/orkes/conductor/client/e2e/SubWorkflowVersionTests.java#L47